Graphs/ChordGraph/index.js

```diff
@@ -153,21 +156,26 @@ export default class ChordGraph extends AbstractGraph {
             .colors(colors);
 
         if(onMarkClick){
-            this.chordDiagram.onSelectedRibbonChange((d) => {
-                const selectedRibbon = this.chordDiagram.selectedRibbon();
-                if(selectedRibbon) {
-                    const { source, destination } = selectedRibbon;
-                    onMarkClick({
-                        [chordSourceColumn]: source,
-                        [chordDestinationColumn]: destination
-                    });
-                } else {
-                    onMarkClick({
-                        [chordSourceColumn]: undefined,
-                        [chordDestinationColumn]: undefined
-                    });
-                }
-            });
+          this.chordDiagram.onSelectedRibbonChange((d) => {
+            const selectedRibbon = this.chordDiagram.selectedRibbon();
+            if (selectedRibbon) {
+              const { source, destination, data } = selectedRibbon;
+              let finalData = [];
+              if (additionalKey) {
+                finalData = data.map((d) => pick(d, additionalKey));
+              }
+              onMarkClick({
+                [chordSourceColumn]: source,
+                [chordDestinationColumn]: destination,
+                data: uniq(finalData)
```

Comment:
@killanch  The query as you expect will not work since as per your request you will have flowid as an array. We will need to have a way of processing that array which should not be the job of the chord diagram. We will keep as array of objects as per my comments above and we will should be able to add more then one key.

Reply:
@natabal Im ok with any format. All I am saying is that, as in my query example above where I use `{{hash}}`, `hash` is expected to be an array of strings.
eventListener should make available `hash` somewhere

Reply:
@killanch What you will need here to be able to evaluate an expression to get the correct string